src/EditorFeatures/CSharpTest2/Recommendations/NotKeywordRecommenderTests.cs

```diff
@@ -0,0 +1,183 @@
+@"var result = e switch
+{
+    $$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAtBeginningOfSwitchStatement()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"switch (e)
+{
+    case $$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInMiddleOfSwitchExpression()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"var result = e switch
+{
+    1 => 2,
+    $$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInMiddleOfSwitchStatement()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"switch (e)
+{
+    case 1:
+    case $$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAtBeginningOfSwitchExpression_AfterOpenParen()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"var result = e switch
+{
+    ($$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInMiddleOfSwitchExpression_AfterOpenParen()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"var result = e switch
+{
+    1 => 2,
+    ($$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInMiddleOfSwitchExpression_ComplexCase()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"var result = e switch
+{
+    1 and ($$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAtBeginningOfSwitchStatement_AfterOpenParen()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(InitializeObjectE +
+@"switch (e)
+{
+    case ($$"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInsideSubpattern()
+        {
+            await VerifyKeywordAsync(
+@"class C
+{
+    public int P { get; }
+
+    void M(C test)
+    {
+        if (test is { P: $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInsideSubpattern_AfterOpenParen()
+        {
+            await VerifyKeywordAsync(
+@"class C
+{
+    public int P { get; }
+
+    void M(C test)
+    {
+        if (test is { P: ($$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInsideSubpattern_AfterOpenParen_Complex()
+        {
+            await VerifyKeywordAsync(
+@"class C
+{
+    public int P { get; }
+
+    void M(C test)
+    {
+        if (test is { P: (1 or $$");
+        }
+
+#endif
+    }
```

Comment:
this set of tests is fantastic.  did you come up with it all yourself?  or did you draw inspiration from some other set of tests?

Reply:
Thank you! I used some of the other keyword recommender tests as a basis of how to write them.